src/api/posts.rs

```diff
@@ -12,15 +13,144 @@ use plume_models::{
+        Connection as Conn,
+        api_tokens::*,
+        apps::*,
+        blogs::*,
+        blog_authors::*,
+        db_conn::{DbConn, DbPool},
+        instance::*,
+        posts::Post,
+        safe_string::SafeString,
+        schema,
+        users::*,
+    };
+    use rocket::http::{Header, ContentType};
+    use serde_json;
+    use test_client;
+
+    #[cfg(feature = "sqlite")]
+    embed_migrations!("migrations/sqlite");
+
+    #[cfg(feature = "postgres")]
+    embed_migrations!("migrations/postgres");
+
+    fn setup_db(conn: &Conn) {
+        embedded_migrations::run(conn).expect("Couldn't run migrations");
+        diesel::delete(schema::instances::table);
+        diesel::delete(schema::blogs::table);
+        diesel::delete(schema::blog_authors::table);
+        diesel::delete(schema::users::table);
+        diesel::delete(schema::posts::table);
+
+        Instance::insert(conn, NewInstance {
+            default_license: "WTFPL".to_string(),
+            local: true,
+            long_description: SafeString::new("This is my instance."),
+            long_description_html: "<p>This is my instance</p>".to_string(),
+            short_description: SafeString::new("My instance."),
+            short_description_html: "<p>My instance</p>".to_string(),
+            name: "My instance".to_string(),
+            open_registrations: true,
+            public_domain: "plu.me".to_string(),
+        });
+        let user = NewUser::new_local(
+            conn,
+            "admin".to_owned(),
+            "The admin".to_owned(),
+            true,
+            "Hello there, I'm the admin".to_owned(),
+            "admin@example.com".to_owned(),
+            "invalid_admin_password".to_owned(),
+        );
+        let blog = Blog::insert(conn, NewBlog::new_local(
+            "MyBlog".to_owned(),
+            "My blog".to_owned(),
+            "Welcome to my blog".to_owned(),
+            Instance::local_id(conn),
+        ));
+        BlogAuthor::insert(conn, NewBlogAuthor {
+            blog_id: blog.id,
+            author_id: user.id,
+            is_owner: true,
+        });
+    }
+
+    fn api_token_for(conn: &Conn, user: User) -> ApiToken {
+        let client_id = random_hex();
+        let client_secret = random_hex();
+        let app = App::insert(conn, NewApp {
+            name: "Test app".to_string(),
+            client_id: client_id,
+            client_secret: client_secret,
+            redirect_uri: None,
+            website: None,
+        });
+
+        ApiToken::insert(conn, NewApiToken {
+            app_id: app.id,
+            user_id: user.id,
+            value: random_hex(),
+            scopes: "write".to_string(),
+        })
+    }
+
+    #[test]
+    fn create() {
+        let client = test_client();
+        let pool = client.rocket().state::<DbPool>().expect("DbPool is not managed");
+        let conn = DbConn(pool.get().expect("Test DB pool error"));
+        setup_db(&*conn);
```

Comment:
anything after `let conn = ...` should be surrounded by a big
```rust
conn.test_transaction::<_, (), _>(|| { 
    [...]
    Ok(())
})
```
to prevent any side effect